README.md

```diff
@@ -1 +1,58 @@
 # Topic Modeling Experiments on Source Code
+
+## Getting Started
+
+_In the following we do not do so, however it is good practice to limit the amount of memory docker containers have access to with the `-m` flag._
+
+Start by cloning the repository, then building the docker image:
+
+```
+git clone https://github.com/src-d/tm-experiments
+```
+
+You should also create a directory for all the data. In the following, we call this `/path/to/data`, and will be using defaults values provided by the container for inputs/outputs.
+
+### Preprocess
+
+Launch Babelfish, install all drivers and check you can see them:
+
+```
+docker run -d --rm --name tmexp_bblfshd --privileged -p 9432:9432 -v /var/lib/bblfshd:/var/lib/bblfshd bblfsh/bblfshd:latest-drivers
+docker exec -it tmexp_bblfshd bblfshctl driver install --all -f
+docker exec -it tmexp_bblfshd bblfshctl driver list
+```
+
+Clone one or multiple repository in a directory, then launch gitbase:
+
+```
+docker run -d --rm --name gitbase -p 3306:3306 --link tmexp_bblfshd:tmexp_bblfshd -e BBLFSH_ENDPOINT=tmexp_bblfshd:9432 -v /path/to/repos:/opt/repos srcd/gitbase:latest
+```
+
+Finally, launch the preprocessing (we give the docker socket in order to be able to relaunch Babelfish, however this is a temporary hack):
+
+```
+docker run --rm -i -v /var/run/docker.sock:/var/run/docker.sock -v /path/to/data:/data --link tmexp_bblfshd:tmexp_bblfshd --link gitbase:gitbase tmexp preprocess -r repo
```

Comment:
Now that we have 3 containers and 3 links, it's probably better to switch to the non-deprecated network instead of link, but since this is already working I'm ok with leaving it as is. Just mentioning it for a future experiment :)

Reply:
I've never used it, how does it work ?

Reply:
Basically you create a network and specify it for each container. For example, in `style-analyzer`, I used:

    docker network create --driver bridge report_0.1.1
    docker run --network report_0.1.1 -d --rm --name report_0.1.1_bblfshd --privileged bblfsh/bblfshd\:v2.11.0 --log-level DEBUG
    docker exec report_0.1.1_bblfshd bblfshctl driver install javascript docker://bblfsh/javascript-driver:v1.2.0
    docker run --network report_0.1.1 -e LOOKOUT_BBLFSHD=ipv4://report_0.1.1_bblfshd:9432 --name report_0.1.1 --entrypoint /bin/bash -it --rm -v /storage/hugo:/storage -v /home/hugo:/home ubuntu:18.04

The advantage is that all containers see each other directly in the network, no need for linking. It's a bit easier to maintain and it's not deprecated (link is). As I said since this works it's not necessary to tweak it right now, I was mentioning it to let you know for the next time :)

Reply:
Oke, no it looks cool - and idk that link as depreciated. Will probably include this in future PR, it makes sense to use it

Reply:
I should mention that it has a “disadvantage”: it persists after all containers have been removed if it is not removed also.

Reply:
ah, hm I guess it's not so bad but yeah got to keep it in mind then